packages/api/portfolioDrafts/application/createApplicationStep.ts

```diff
@@ -44,8 +50,27 @@ export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayPr
   if (!isApplicationStep(requestBody)) {
     return REQUEST_BODY_INVALID;
   }
+
+  const { applications, operators } = requestBody;
+  if (applications.filter((app) => !isApplication(app)).length) {
+    return REQUEST_BODY_INVALID;
+  }
+
+  const appEnvs: Array<Environment> = applications.flatMap((app) => app.environments);
+  const invalidEnvs = appEnvs.filter((env) => !isEnvironment(env));
+  if (invalidEnvs.length) {
+    return REQUEST_BODY_INVALID;
+  }
+
+  const appOperators: Array<AppEnvOperator> = applications.flatMap((app) => app.operators);
+  const envOperators: Array<AppEnvOperator> = appEnvs.flatMap((env) => env.operators);
```

Comment:
I am curious why the types of these cannot be inferred. The fact it's an array feels obvious so the real issue seems to be arround `.environments` and `.operators`. Any insight to what error message (if any) you get when removing the type info here?